### PR TITLE
Store some more things in the database

### PR DESCRIPTION
**Pull request by TobiasFella**
It's unused for now but we'll need it for device/message verification

### CODE REVIEW

File: lib/database.cpp [hunk 1]
```diff
@@ -83,7 +83,7 @@ void Database::migrateTo1()
     transaction();
     execute(QStringLiteral("CREATE TABLE accounts (pickle TEXT);"));
     execute(QStringLiteral("CREATE TABLE olm_sessions (senderKey TEXT, sessionId TEXT, pickle TEXT);"));
-    execute(QStringLiteral("CREATE TABLE inbound_megolm_sessions (roomId TEXT, senderKey TEXT, sessionId TEXT, pickle TEXT);"));
+    execute(QStringLiteral("CREATE TABLE inbound_megolm_sessions (roomId TEXT, senderKey TEXT, sessionId TEXT, ed25519Key TEXT, pickle TEXT);"));
```
Comment (CarlSchwan) [inline lib/database.cpp]: you probably need a migration here?

Comment (TobiasFella) [reply lib/database.cpp]: I wanted to avoid that, but yes, you're right...

**Review by CarlSchwan — APPROVED** [review]
Code looks good 👍

**Comment by TobiasFella** [comment]
The commits are only related in that they all change the database schema, by doing both in one merge request we don't need another migration

File: lib/database.cpp [hunk 2]
```diff
@@ -94,6 +95,16 @@ void Database::migrateTo1()
     commit();
 }
 
+void Database::migrateTo2()
+{
+    qCDebug(DATABASE) << "Migrating database to version 2";
+    transaction();
+    execute(QStringLiteral("ALTER TABLE inbound_megolm_sessions ADD ed25519Key TEXT"));
+    execute(QStringLiteral("ALTER TABLE olm_sessions ADD lastReceived TEXT"));
```
Comment (CarlSchwan) [inline lib/database.cpp:103]:
```suggestion
    execute(QStringLiteral("ALTER TABLE olm_sessions ADD lastReceived TEXT"));
    
    // Add indexes for improving queries speed on larger database
    execute(QStringLiteral("CREATE INDEX sessions_session_idx ON olm_sessions(sessionId)"));
    execute(QStringLiteral("CREATE INDEX outbound_room_idx ON outbound_megolm_sessions(roomId)"));
    execute(QStringLiteral("CREATE INDEX inbound_room_idx ON inbound_megolm_sessions(roomId)"));
    execute(QStringLiteral("CREATE INDEX group_session_idx ON group_session_record_index(roomId, sessionId, i)"));
```

warning: not tested

File: lib/connection.cpp [hunk 3]
```diff
@@ -2150,9 +2152,9 @@ UnorderedMap<std::pair<QString, QString>, QOlmInboundGroupSessionPtr> Connection
     return database()->loadMegolmSessions(room->id(), picklingMode());
 }
 
-void Connection::saveMegolmSession(Room* room, const QString& senderKey, QOlmInboundGroupSession* session)
+void Connection::saveMegolmSession(Room* room, const QString& senderKey, QOlmInboundGroupSession* session, const QString& ed25519Key)
 {
-    database()->saveMegolmSession(room->id(), senderKey, session->sessionId(), session->pickle(picklingMode()));
+    database()->saveMegolmSession(room->id(), senderKey, session->sessionId(), ed25519Key, session->pickle(picklingMode()));
```
Comment (KitsuneRal) [inline lib/connection.cpp:2162]: What if `ed25519Key` comes empty? Either a validation in either of the two `saveMegolmSession()` methods, or a database constraint (and a check that the query completed successfully) would be worthwhile. Or is there already a check that I missed?

**Review by KitsuneRal — CHANGES_REQUESTED** [review]

File: lib/database.cpp (hunk 2 above)
Comment (TobiasFella) [reply lib/database.cpp:103]: it compiles and executes fine. i can't judge if it's doing what it's intended to do

Comment (CarlSchwan) [reply lib/database.cpp:103]: It's adding index to speed up the select where queries by building a B-Tree

File: lib/connection.cpp (hunk 3 above)
Comment (TobiasFella) [reply lib/connection.cpp:2162]: done

**Review by KitsuneRal — APPROVED** [review]

**Pull request merged**